src/excmds.ts

```diff
@@ -4345,35 +4347,231 @@ const KILL_STACK: Element[] = []
+                                newOpenMode = OpenMode.ImagesTab
+                                break
+                            case "k":
+                                newOpenMode = OpenMode.Kill
+                                break
+                            case "K":
+                                newOpenMode = OpenMode.KillTridactyl
+                                break
+                            case "z":
+                                newOpenMode = OpenMode.Scroll
+                                break
+                            case "s":
+                                newOpenMode = OpenMode.SaveResource
+                                break
+                            case "S":
+                                newOpenMode = OpenMode.SaveImage
+                                break
+                            case "a":
+                                newOpenMode = OpenMode.SaveAsResource
+                                break
+                            case "A":
+                                newOpenMode = OpenMode.SaveAsImage
+                                break
+                            case ";":
+                                newOpenMode = OpenMode.Semicolon
+                                break
+                            case "r":
+                                newOpenMode = OpenMode.TTSRead
+                                break
+                            case "P":
+                                newOpenMode = OpenMode.YankAlt
+                                break
+                            case "#":
+                                newOpenMode = OpenMode.YankAnchor
+                                break
+                            case "y":
+                                newOpenMode = OpenMode.YankLink
+                                break
+                            case "p":
+                                newOpenMode = OpenMode.YankText
+                                break
+                            default:
+                                logger.warning(`unknown flag -${flag}`)
+                                break
+                        }
 
-    // extract flags
-    // Note: we need to process 'pipe' separately because it could be interpreted as -p -i -e otherwise
-    const pipeIndex = option.indexOf("pipe")
-    if (pipeIndex >= 0) {
-        option = option.slice(0, pipeIndex) + option.slice(pipeIndex + 4)
-    }
+                        if (newOpenMode !== undefined) {
+                            if (openMode !== OpenMode.Default) {
+                                // Notify that multiple open modes doesn't make sense
+                                logger.warning("multiple open mode flags specified, overriding the previous ones")
+                            }
 
-    // Hacky fix for #1374 - join rapid-hinted yanks
-    if (option === "-qy") {
-        run_exstr('composite hint -qpipe a href | js -p JS_ARG.join(" ") | yank')
```

Comment:
I'm somewhat reluctant to lose this workaround since it was specifically requested. Can't we generalise it for all yank modes?

Reply:
I forgot I wanted to look at this issue before pushing. As a first step I'll replicate the behavior only for -y, but maybe this should be the case for all types of yanks?

Reply:
Fixed by https://github.com/tridactyl/tridactyl/pull/3534/commits/be5a66e046530f078c5bfe38bfa3ab09c7efe475. All yanks will put their results in the clipboard as a newline-delimited string. This works for rapid (-q) and immediate mode (-!). The key was that the hintPage function already resolves to the "onSelect" callback value, so it was just a matter of processing the result with `.then`